elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+            except FileNotFoundError:
+                return 'The instructor did not include a reference solution. Try contacting them for the solution implementation?'
+
+        permutationMode = pl.get_string_attrib(element, 'permutation-mode', 'html-order')
+        permutationMode = ' in any order' if permutationMode == 'any' else 'in the specified order'
+
+        if answerName in data['correct_answers']:
+            html_params = {
+                'true_answer': True,
+                'question_solution': str(data['correct_answers'][answerName]['correct_answers']),
+                'permutationMode': permutationMode
+            }
+            with open('pl-drag-drop-element.mustache', 'r', encoding='utf-8') as f:
+                html = chevron.render(f, html_params).strip()
+            return html
+        else:
+            return ''
+
+
+def prepare(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answerName = pl.get_string_attrib(element, 'answers-name')
+
+    mcq_options = []
+    correct_answers = []
+    correct_answers_indent = []
+
+    isShuffle = pl.get_string_attrib(element, 'shuffle-options', 'false')  # default to FALSE, no shuffling unless otherwise specified
+
+    for html_tags in element:
+        if html_tags.tag == 'pl-answer':
+            # CORRECT is optional for backward compatibility
+            # pl.check_attribs(html_tags, required_attribs=[''], optional_attribs=['correct', 'ranking', 'indent'])
+            mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+        # BACKWARD COMPATIBILITY CODE
+        if html_tags.tag == 'pl-distractor':
+            mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+
+    if isShuffle == 'true':
+        random.shuffle(mcq_options)
+
+    for html_tags in element:
+        if html_tags.tag == 'pl-answer':
+            isCorrect = pl.get_string_attrib(html_tags, 'correct', 'true')  # default to true, for backward compatibility
+            answerIndent = pl.get_string_attrib(html_tags, 'indent', '-1')  # get answer indent, and default to -1 (indent level ignored)
+            if isCorrect.lower() == 'true':
+                # add option to the correct answer array, along with the correct required indent
+                correct_answers.append(str.strip(html_tags.text))
+                correct_answers_indent.append(answerIndent)
+
+    data['params'][answerName] = mcq_options
+    data['correct_answers'][answerName] = {'correct_answers': correct_answers,
+                                           'correct_answers_indent': correct_answers_indent}
+
+
+def parse(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answerName = pl.get_string_attrib(element, 'answers-name')
+
+    temp = answerName
+    temp += '-input'  # this is how the backend is written
```

Comment:
what does this comment mean? Its not very clear.

Reply:
this comment was revised in bcadc9f